File: pset2/readability/readability.c
//Import the libraries needed
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

//Declare my functions
int count_letters(string t);
int count_words(string t);
int count_sentences(string t);
float index(int l, int w, int s);

//Runs the main program
int main(void)
{
    string text = get_string("input text: ");
    float i = index(count_letters(text), count_words(text), count_sentences(text));

    if (i < 16 && i >= 1)
    {
        printf("Grade %i\n", (int) round(i));
    }
    else if (i >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }

    return 0;
}

//Creates function to count the amount of letters
int count_letters(string t)
{
    int letters_count = 0;

    for (int i = 0; i < strlen(t); i++)
    {
        if (islower(t[i]) != 0 || isupper(t[i]) != 0)
        {
            letters_count++;
        }
        else
        {
            continue;
        }
    }
    return letters_count;
}

//Creates function to count the amount of words
int count_words(string t)
{
    int words_count = 0;

    for (int i = 0; i < strlen(t); i++)
    {
        if (isblank(t[i]) != 0)
        {
            words_count++;
        }
    }
    return words_count + 1;
}

//Creates function to count the amount of sentences
int count_sentences(string t)
{
    int sentences_count = 0;

    for (int i = 0; i < strlen(t); i++)
    {
        if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {

            sentences_count++;
        }
    }
    return sentences_count;
}

//Creates function to return Coleman-Liau index
float index(int l, int w, int s)
{
    float a = 100 / (float) w;
    float index = 0.0588 * ((float) l * a) - 0.296 * ((float) s * a) - 15.8;
    return index;
}

File: pset6/readability/readability.py
from cs50 import get_string

# Runs the main program


def main():
    text = get_string("input text: ")
    i = index(count_letters(text), count_words(text), count_sentences(text))

    if i < 16 and i >= 1:
        print("Grade", int(round(i)))
    elif i >= 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")
    return 0

# Creates function to count the amount of letters


def count_letters(t):
    letters_count = 0
    for i in range(len(t)):
        if t[i].islower() is True:
            letters_count += 1
        elif t[i].isupper() is True:
            letters_count += 1
        else:
            continue
    return letters_count

# Creates function to count the amount of words


def count_words(t):
    words_count = 0
    for i in range(len(t)):
        if t[i].isspace() is True:
            words_count += 1
    return words_count + 1

# Creates function to count the amount of sentences


def count_sentences(t):
    sentences_count = 0
    for i in range(len(t)):
        if t[i] is '.' or t[i] is '!' or t[i] is '?':
            sentences_count += 1
    return sentences_count

# Creates function to return Coleman-Liau index


def index(l, w, s):
    a = 100 / w
    index = 0.0588 * (l * a) - 0.296 * (s * a) - 15.8
    return index


main()